File: lib/swig/cs/VersionListComboBox.cs
/*
 * Copyright (c) 2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

using System.Windows.Forms;

namespace PKMN
{
    namespace Forms
    {
        public class VersionListComboBox: ComboBox
        {
            public VersionListComboBox(): base()
            {
                StringVector version_vec = new StringVector();
                Lists.getVersionList(version_vec);
                for(int i = 0; i < version_vec.Count; i++)
                {
                    Items.Add(version_vec[i]);
                }
            }
        }
    }
}

File: lib/swig/java/VersionListComboBox.java
/*
 * Copyright (c) 2013-2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package org.nc.PKMN;

import javax.swing.*;

public class VersionListComboBox extends JComboBox<String>
{
    public VersionListComboBox(int gen)
    {
        super();

        StringVector versionVec = new StringVector();
        Lists.getVersionList(versionVec);
        for(int i = 0; i < versionVec.size(); i++)
        {
            addItem(versionVec.get(i));
        }

        setEditable(false);
    }
}
